include/boost/geometry/formulas/quarter_meridian.hpp

```diff
@@ -0,0 +1,102 @@
+// Boost.Geometry
+
+// Copyright (c) 2018 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_QUARTER_MERIDIAN_HPP
+#define BOOST_GEOMETRY_FORMULAS_QUARTER_MERIDIAN_HPP
+
+#include <boost/geometry/core/radius.hpp>
+#include <boost/geometry/core/tag.hpp>
+#include <boost/geometry/core/tags.hpp>
+
+#include <boost/geometry/algorithms/not_implemented.hpp>
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DISPATCH
+namespace formula_dispatch
+{
+
+template <typename ResultType, typename Geometry, typename Tag = typename tag<Geometry>::type>
+struct quarter_meridian
+    : not_implemented<Tag>
+{};
+
+template <typename ResultType, typename Geometry>
+struct quarter_meridian<ResultType, Geometry, srs_spheroid_tag>
+{
+    //https://en.wikipedia.org/wiki/Meridian_arc#Generalized_series
+    static inline ResultType apply(Geometry const& geometry)
+    {
+        //order 8 expansion
+        ResultType const C[] =
+        {
+            1073741824,
+            268435456,
+            16777216,
+            4194304,
+            1638400,
+            802816,
+            451584,
+            278784,
+            184041
+        };
+
+        ResultType const c2 = 2;
+        ResultType const c4 = 4;
+        ResultType const f = formula::flattening<ResultType>(geometry);
+        ResultType const n = f / (c2 - f);
+        ResultType const ab4 = (get_radius<0>(geometry)
+                                + get_radius<2>(geometry)) / c4;
+        return geometry::math::pi<ResultType>() * ab4 *
+                 horner_evaluate(n*n, C, C+8) / C[0];
```

Comment:
So this function will calculate 8 nested levels of products of products or products, etc. My guess is that the error propagation will be big. Is this really a problem? Would representing this series in a "typical" way improve the accuracy of the result, i.e. as sum of ratios multiplied by x^i?

Reply:
First, using horner rule (i.e. nested evaluation) should be faster than the standard evaluation, the first requires 2n operations while the second 3n-1 (if powers are computed iteratively). Second, horner seems to me more accurate since it uses less operations. Third, if this function is placed somewhere in geometry::math or in utilities then it could be used wherever we have to evaluate some polynomial. This will make the code shorter and more elegant than implementing manually polynomial evaluations in standard form when it is needed.